state/status.go

```diff
@@ -19,6 +19,165 @@ import (
 	"github.com/juju/juju/status"
 )
 
+// ModelStatus holds all the current status values for a given model
+// and offers accessors for the various parts of a model.
+type ModelStatus struct {
+	model *Model
+	docs  map[string]statusDocWithID
+}
+
+// LoadModelStatus retrieves all the status documents for the model
+// at once. Used to primarily speed up status.
+func (m *Model) LoadModelStatus() (*ModelStatus, error) {
```

Comment:
I think @jameinel's right - it would be a pity to dramatically slow down status with a filter pattern on a large model. Would it be hard to apply the filters here?